code/javascript/example.js

```diff
@@ -1,27 +1,41 @@
 // npm install --save neo4j-driver
 // node example.js
-const neo4j = require('neo4j-driver');
-const driver = neo4j.driver('neo4j+s://demo.neo4jlabs.com:7687',
-                  neo4j.auth.basic('mUser', 's3cr3t'), 
-                  {});
+const neo4j = require("neo4j-driver");
 
-const query =
-  `
-  MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) RETURN a.name as actorName
-  `;
+async function main() {
+  const driver = neo4j.driver("neo4j+s://demo.neo4jlabs.com:7687", neo4j.auth.basic("mUser", "s3cr3t"));
 
-const params = {"movieTitle": "The Matrix"};
+  try {
+    const products = await getDairyProducts(driver)
```

Comment:
no domain specific methods that don't work with the templating
why not just an inline closure instead of the full ceremony?

Reply:
in this simple example we can close driver and session together